crates/virtio-queue/src/lib.rs

```diff
@@ -1493,11 +1567,35 @@ mod tests {
         let vq = MockSplitQueue::new(m, 16);
 
         let mut q = vq.create_queue(m);
-        q.state.size = 8;
-        q.state.ready = true;
+        q.set_size(8);
+        // The address set by `MockSplitQueue` for the descriptor table is DEFAULT_DESC_TABLE_ADDR,
+        // so let's change it for testing the reset.
+        q.set_desc_table_address(Some(0x5000), None);
+        // Same for `event_idx_enabled`, `next_avail` `next_used` and `signalled_used`.
+        q.set_event_idx(true);
+        q.set_next_avail(2);
+        q.add_used(1, 200).unwrap();
+        q.state.signalled_used = Some(Wrapping(15));
+        assert_eq!(q.state.size, 8);
+        // `create_queue` also marks the queue as ready.
+        assert_eq!(q.state.ready, true);
+        assert_ne!(q.state.desc_table, GuestAddress(DEFAULT_DESC_TABLE_ADDR));
+        assert_ne!(q.state.avail_ring, GuestAddress(DEFAULT_AVAIL_RING_ADDR));
+        assert_ne!(q.state.used_ring, GuestAddress(DEFAULT_USED_RING_ADDR));
+        assert_ne!(q.state.next_avail, Wrapping(0));
+        assert_ne!(q.state.next_used, Wrapping(0));
+        assert_ne!(q.state.signalled_used, None);
```

Comment:
Why do we check these before and after the reset? They seem to have the same values.

Reply:
I am using `assert_ne!` before the reset.